sippy-ng/src/releases/PayloadTestFailures.js

```diff
@@ -0,0 +1,281 @@
+      },
+    },
+  ]
+
+  const [payload = props.payload, setPayload] = useQueryParam(
+    'payload',
+    StringParam
+  )
+  const [fetchError, setFetchError] = React.useState('')
+  const [isLoaded, setLoaded] = React.useState(false)
+  const [rows, setRows] = React.useState([])
+
+  const [filterModel = props.filterModel, setFilterModel] = useQueryParam(
+    'filters',
+    SafeJSONParam
+  )
+
+  const [sortField = props.sortField, setSortField] = useQueryParam(
+    'sortField',
+    StringParam
+  )
+  const [sort = props.sort, setSort] = useQueryParam('sort', StringParam)
+
+  const [pageSize = props.pageSize, setPageSize] = useQueryParam(
+    'pageSize',
+    NumberParam
+  )
+
+  const requestSearch = (searchValue) => {
+    const currentFilters = filterModel
+    currentFilters.items = currentFilters.items.filter(
+      (f) => f.columnField !== 'name'
+    )
+    currentFilters.items.push({
+      id: 99,
+      columnField: 'name',
+      operatorValue: 'contains',
+      value: searchValue,
+    })
+    setFilterModel(currentFilters)
+  }
+
+  const addFilters = (filter) => {
+    const currentFilters = filterModel.items.filter((item) => item.value !== '')
+
+    filter.forEach((item) => {
+      if (item.value && item.value !== '') {
+        currentFilters.push(item)
+      }
+    })
+    setFilterModel({
+      items: currentFilters,
+      linkOperator: filterModel.linkOperator || 'and',
+    })
+  }
+
+  const updateSortModel = (model) => {
+    if (model.length === 0) {
+      return
+    }
+
+    if (sort !== model[0].sort) {
+      setSort(model[0].sort)
+    }
+
+    if (sortField !== model[0].field) {
+      setSortField(model[0].field)
+    }
+  }
+
+  const fetchData = () => {
+    let queryString = ''
+    if (filterModel && filterModel.items.length > 0) {
+      queryString +=
+        '&filter=' + safeEncodeURIComponent(JSON.stringify(filterModel))
+    }
+
+    Promise.all([
+      fetch(
+        `${process.env.REACT_APP_API_URL}/api/payloads/test_failures?payload=${payload}` +
+          queryString
+      ),
+    ])
+      .then(([analysis]) => {
+        if (analysis.status !== 200) {
+          throw new Error('server returned ' + analysis.status)
+        }
+
+        return Promise.all([analysis.json()])
+      })
+      .then(([analysis]) => {
+        setRows(analysis)
+        setLoaded(true)
+      })
+      .catch((error) => {
+        setFetchError(
+          'Could not retrieve payload analysis for ' +
+            props.payload +
+            ': ' +
+            error
+        )
+      })
+  }
+
+  useEffect(() => {
+    fetchData()
+  }, [filterModel, sort, sortField])
+
+  if (fetchError !== '') {
+    return <Alert severity="error">Failed to load data, {fetchError}</Alert>
+  }
+
+  if (!isLoaded) {
+    return <p>Loading...</p>
+  }
+
+  if (rows.length === 0) {
+    return <Typography variant="h5">Analysis not found.</Typography>
+  }
+
+  console.log('hello world')
```

Comment:
nit: not sure you need it -- but harmless.

Reply:
My bad, thanks!

https://github.com/openshift/sippy/pull/1148